components/brave_webtorrent/extension/components/torrentViewerHeader.tsx

```diff
@@ -19,6 +19,63 @@ interface Props {
   onStopDownload: (tabId: number) => void
 }
 
+let alertShown = false
+const remoteProtocols = ['http:', 'https:', 'ftp:', 'sftp:', 'ws:', 'wss:']
+const isTorrentFile = (filename = '', mimeType = '', pathname = '') => {
+  // same logic as IsTorrentFile from browser/webtorrent_util.cc
+  if (mimeType === 'application/x-bittorrent') {
```

Comment:
I think the attacker can serve the malicious file (e.g. `attack.bat`) with a `application/x-bittorrent` mimetype and it will be allowed to download. Is that right? One thing that could make this better is to force a `.torrent` extension on all files downloaded with this mechanism.

But we don't enforce any restrictions like this on normal file downloads, so I don't know if this is really a reasonable task for the webtorrent extension to handle given that we don't protect against normal sites using this trick.

Reply:
@feross i believe so; the logic in https://github.com/brave/brave-core/blob/master/components/brave_webtorrent/browser/webtorrent_util.cc#L82-L84 for determining whether to intercept a request with webtorrent considers a `application/x-bittorrent` MIME type to be a torrent regardless of the file type. i wasn't sure if legit torrent files are ever served with this MIME type but not the .torrent extension. if that's not the case, i think we can probably change this check to "does this file end with .torrent AND is the MIME type either 'application/octet-stream' or 'application/x-bittorrent'"

Reply:
apparently windows will execute a .BAT file regardless of MIME type headers if the extension is .BAT. i'm going to change this check to just check if the filename ends with .torrent